app/src/main/java/com/example/drawingApp/DialogObject.kt

```diff
@@ -0,0 +1,119 @@
+        onDialogSubmission: (String) -> Unit
+    ) {
+        val inflater = LayoutInflater.from(c)
+        val view = inflater.inflate(R.layout.layer_dialog, null)
+        val layerField = view.findViewById<EditText>(R.id.newLayerField)
+        val errorText = view.findViewById<TextView>(R.id.dialogError)
+        val layerNames = mutableListOf<String>()
+        for (x in model.layerNames) layerNames.add(x.text)
+        var layerCounter = 1
+
+
+        //find the first available layer name not taken
+        while (layerNames.contains(
+                c.resources.getString(
+                    R.string.new_layer_hint,
+                    layerCounter
+                )
+            )
+        ) {
+            layerCounter += 1
+        }
+        layerField.setText(c.resources.getString(R.string.new_layer_hint, layerCounter))
+
+        //needs to be after the set text or else the textchanged listener will overwrite the previous text needed for rotations
+        layerField.addTextChangedListener {
+            val fieldString = it.toString().trim()
+            model.currentLayerCreationDialogText = fieldString
+            errorText.setText(R.string.charCountError)
+            errorText.isVisible = fieldString.length > MainActivity.MAX_CHAR_LEN
+        }
+
+        val dialog = AlertDialog.Builder(c).apply {
+            setView(view)
+        }.create()
+
+        view.findViewById<TextView>(R.id.newLayerCancel).setOnClickListener {
+            dialog.dismiss()
+        }
+
+        view.findViewById<TextView>(R.id.newLayerOk).setOnClickListener {
+            val fieldString = layerField.text.toString().trim()
+            if (fieldString.length > MainActivity.MAX_CHAR_LEN) {
+                errorText.visibility = View.VISIBLE
+            } else if (layerNames.contains(fieldString)) {
+                errorText.setText(R.string.layerExistsError)
+                errorText.visibility = View.VISIBLE
+            } else {
+                layerNames.add(fieldString)
+                onDialogSubmission(fieldString)
+                //model.layerNames.add(LayerViewModel(fieldString))
+                Toast.makeText(c, fieldString, Toast.LENGTH_SHORT)
+                    .show()
+                dialog.dismiss()
+            }
+        }
+        //this is here so that in the case that the dialogue was previously open
+        //we will immediately know to set the text to what it was previously
+        //the reason it is here, is so that we avoid overriding the correct text
+        if (model.isLayerCreationDialogOpen) {
+            layerField.setText(model.currentLayerCreationDialogText)
+        }
+        model.isLayerCreationDialogOpen = true
+        dialog.setOnDismissListener { model.isLayerCreationDialogOpen = false }
+        dialog.show()
+    }
+
+    fun showEditLayerDialog(c: Context, model: DrawingViewModel) {
+        val data = model.layerNames
+        lateinit var myAdapter: EditLayerAdapter
```

Comment:
is this a lateinit because it needs to be? or was it possible to do this after you set the view variable? I need to do this still so I will come back to this

Reply:
yes, it needs to be a lateinit to reference the adapter in the lambdas.